encode/src/main/scala/org/broadinstitute/monster/etl/encode/extract/EncodeIngestExtract.scala

```diff
@@ -0,0 +1,148 @@
+          EncodeEntity.File.entryName,
+          EncodeEntity.File.entityType
+        )(fileIDParams)
+      )
+      filteredFiles.saveAsJsonFile(
+        s"${parsedArgs.outputDir}/${EncodeEntity.File.entryName}.json"
+      )
+
+      val auditEntryName: String = "Audits"
+      val auditIDParams = encodeExtractions.getIDParams(
+        auditEntryName,
+        referenceField = "@id",
+        manyReferences = false
+      )(filteredFiles)
+      val transformedAudits = transformAudits(auditEntryName)(
+        encodeExtractions.extractIDParamEntities(auditEntryName, entityType = "File")(
+          auditIDParams
+        )
+      )
+      transformedAudits.saveAsJsonFile(s"${parsedArgs.outputDir}/$auditEntryName.json")
+
+      val replicateEntryName: String = "Replicates"
+      val replicateIDParams = encodeExtractions.getIDParams(
+        replicateEntryName,
+        referenceField = "replicates",
+        manyReferences = true
+      )(experiments)
+      val replicates = encodeExtractions.extractIDParamEntities(
+        replicateEntryName,
+        entityType = "Replicate"
+      )(replicateIDParams)
+      replicates.saveAsJsonFile(s"${parsedArgs.outputDir}/$replicateEntryName.json")
+
+      val libraryIDParams = encodeExtractions.getIDParams(
+        EncodeEntity.Library.entryName,
+        referenceField = "library",
+        manyReferences = false
+      )(replicates)
+      val libraries = encodeExtractions.extractIDParamEntities(
+        EncodeEntity.Library.entryName,
+        EncodeEntity.Library.entityType
+      )(libraryIDParams)
+      libraries.saveAsJsonFile(
+        s"${parsedArgs.outputDir}/${EncodeEntity.Library.entryName}.json"
+      )
+
+      val sampleIDParams = encodeExtractions.getIDParams(
+        EncodeEntity.Biosample.entryName,
+        referenceField = "biosample",
+        manyReferences = false
+      )(libraries)
+      val samples = encodeExtractions.extractIDParamEntities(
+        EncodeEntity.Biosample.entryName,
+        EncodeEntity.Biosample.entityType
+      )(sampleIDParams)
+      samples.saveAsJsonFile(
+        s"${parsedArgs.outputDir}/${EncodeEntity.Biosample.entryName}.json"
+      )
+
+      val donorIDParams = encodeExtractions.getIDParams(
+        EncodeEntity.Donor.entryName,
+        referenceField = "donor",
+        manyReferences = false
+      )(samples)
+      val donors = encodeExtractions.extractIDParamEntities(
+        EncodeEntity.Donor.entryName,
+        EncodeEntity.Donor.entityType
+      )(donorIDParams)
+      donors.saveAsJsonFile(
+        s"${parsedArgs.outputDir}/${EncodeEntity.Donor.entryName}.json"
+      )
+
+      // waitUntilDone() throws error on failure
+      IO(pipelineContext.close().waitUntilDone()).as(ExitCode.Success)
+    }
+  }
+
+  def filterFiles(
+    entryName: String
+  ): SCollection[JsonObject] => SCollection[JsonObject] =
+    _.transform(s"filterFiles - $entryName") {
+      _.filter(
```

Comment:
Tip: `_.filter { ... }` will format more nicely than `_.filter(...)` when the body is multi-line.